### PR TITLE
update to use java 17

### PR DESCRIPTION
**Pull request by gkwan-ibm**
address issue #204

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -34,51 +34,19 @@ You will learn how to build a RESTful microservice on Liberty with Jakarta EE an
...
- - Set up https://docs.docker.com/docker-for-mac/#kubernetes[Docker for Mac^].
-
-After you complete the Docker setup instructions for your operating system, ensure that Kubernetes (not Swarm) is selected as the orchestrator in Docker Preferences.
---
-
-[.tab_content.linux_section]
---
-You will use `Minikube` as a single-node Kubernetes cluster that runs locally in a virtual machine.
-Make sure you have `kubectl` installed. If you need to install `kubectl`, see the https://kubernetes.io/docs/tasks/tools/install-kubectl/#install-kubectl-on-linux[kubectl installation instructions^].
+You will use `Minikube` as a single-node Kubernetes cluster that runs locally in a virtual machine. Make sure you have `kubectl` installed. If you need to install `kubectl`, see the https://kubernetes.io/docs/tasks/tools/install-kubectl/#install-kubectl-on-linux[kubectl installation instructions^].
```
Comment (ramkumar-k-9286) [inline README.adoc]:
```suggestion
You will use `Minikube` as a single-node Kubernetes cluster that runs locally in a virtual machine. Make sure that you have `kubectl` installed. If you need to install `kubectl`, see the https://kubernetes.io/docs/tasks/tools/install-kubectl/#install-kubectl-on-linux[kubectl installation instructions^].
```

File: README.adoc [hunk 2]
```diff
@@ -129,31 +97,59 @@ On that page, enter the following properties in the **Create a starter applicati
...
-* Under Java EE/Jakarta EE Version select: `10`
+* Under Java SE Version select: `17`
+* Under Java EE/Jakarta EE Version select: `10.0`
 * Under MicroProfile Version select: `6.0`
 
 ifndef::cloud-hosted[]
 Then, click `Generate Project`, which downloads the starter project as `inventory.zip` file. 
 
-Next, extract the `inventory.zip` file on your system. Move the contents of this extracted `inventory` directory to the `start` directory of this project, which is located at the following path: `guide-liberty-deepdive/start/inventory`
+Next, extract the `inventory.zip` file on your system. Move the contents of this extracted `inventory` directory to the `start` directory of this project, which is located at the following path: `guide-liberty-deepdive/start/inventory`.
```
Comment (ramkumar-k-9286) [inline README.adoc]:
```suggestion
Next, extract the `inventory.zip` file on your system. Move the contents of this extracted `inventory` directory to the `start` directory of this project, which is at the following path: `guide-liberty-deepdive/start/inventory`.
```

File: README.adoc [hunk 3]
```diff
@@ -203,12 +199,7 @@ While the Liberty instance starts up, various messages display in your command-l
 [INFO] [AUDIT] CWWKF0011I: The server defaultServer is ready to run a smarter planet.
 ----
 
-When you need to stop the Liberty instance, press `CTRL+C` in the command-line session where you ran the Liberty, or run the `liberty:stop` goal from the `start/inventory` directory in another command-line session:
-
-[role='command']
-```
-mvn liberty:stop
-```
+When you need to stop the Liberty instance, press `CTRL+C` in the command-line session where you ran the Liberty.
```
Comment (ramkumar-k-9286) [inline README.adoc:202]: just to confirm is it `CTRL+C` or is it `CTRL+c` or press `ctrl+c` ?

File: README.adoc [hunk 4]
```diff
@@ -531,7 +528,7 @@ server.xml
 include::finish/module-config/src/main/liberty/config/server.xml[]
 ----
 
-Add variables for the [hotspot=httpPortVariable file=0]`HTTP` port, [hotspot=httpsPortVariable file=0]`HTTPS` port, and the [hotspot=contextRootVariable file=0]`context root` to the `server.xml` configuration file. Change the [hotspot=editedHttpEndpoint file=0]`httpEndpoint` element to reflect the new `default.http.port` and `default.http.port` variables and change the [hotspot=editedContextRoot file=0]`contextRoot` to use the new `default.context.root` variable too.
+Add variables for the [hotspot=httpPortVariable file=0]`HTTP` port, [hotspot=httpsPortVariable file=0]`HTTPS` port, and the [hotspot=contextRootVariable file=0]`context root` to the `server.xml` configuration file. Change the [hotspot=editedHttpEndpoint file=0]`httpEndpoint` element to reflect the new `http.port` and `http.port` variables and change the [hotspot=editedContextRoot file=0]`contextRoot` to use the new `context.root` variable too.
```
Comment (ramkumar-k-9286) [inline README.adoc]:
```suggestion
Add variables for the [hotspot=httpPortVariable file=0]`HTTP` port, [hotspot=httpsPortVariable file=0]`HTTPS` port, and the [hotspot=contextRootVariable file=0]`context root` to the `server.xml` configuration file. Change the [hotspot=editedHttpEndpoint file=0]`httpEndpoint` element to reflect the new `http.port` and `http.port` variable and change the [hotspot=editedContextRoot file=0]`contextRoot` to use the new `context.root` variable too.
```

File: README.adoc [hunk 5]
```diff
@@ -1219,7 +1229,9 @@ include::finish/system/src/main/webapp/META-INF/microprofile-config.properties[]
...
 
-The [hotspot=keyStore file=0]`keyStore` element is used to define the repository of security certificates used for SSL encryption. The `id` attribute is a unique configuration ID that is set to `defaultKeyStore`. The `password` attribute is used to load the keystore file, and its value can be stored in clear text or encoded form. To learn more about other attributes, see the https://openliberty.io/docs/latest/reference/config/keyStore.html#keyStore.html[keyStore^] attribute documentation. 
+The [hotspot=keyStore file=0]`keyStore` element is used to define the repository of security certificates used for SSL encryption. The `id` attribute is a unique configuration ID that is set to `guideKeyStore`. The `password` attribute is used to load the keystore file, and its value can be stored in clear text or encoded form. To learn more about other attributes, see the https://openliberty.io/docs/latest/reference/config/keyStore.html#keyStore.html[keyStore^] attribute documentation.
```
Comment (ramkumar-k-9286) [inline README.adoc]:
```suggestion
The [hotspot=keyStore file=0]`keyStore` element is used to define the repository of security certificates used for SSL encryption. The `id` attribute is a unique configuration ID that is set to `guideKeyStore`. The `password` attribute is used to load the keystore file, and its value can be stored in clear text or encoded form. To learn more about other attributes, see the https://openliberty.io/docs/latest/reference/config/keyStore.html#keyStore.html[keyStore attribute documentation^]. 
```

File: README.adoc [hunk 6]
```diff
@@ -2051,118 +2063,100 @@ Tests run: 3, Failures: 0, Errors: 0, Skipped: 0
...
+Start your Kubernetes cluster.
+
+Run the following command from a command-line session:
+
+[role=command]
+```
+minikube start
+```
+
+Next, validate that you have a healthy {kube} environment by running the following command from the active command-line session.
```
Comment (ramkumar-k-9286) [inline README.adoc]: Should we be saying " healthy {kube} environment " or " healthy kubernetes environment " ?

File: README.adoc [hunk 7]
```diff
@@ -2051,118 +2063,100 @@ Tests run: 3, Failures: 0, Errors: 0, Skipped: 0
...
+
 Now that the containerized application is built and tested, deploy it to a local Kubernetes cluster. 
 
 === Installing the Open Liberty Operator 
 
 ifndef::cloud-hosted[]
 Install the Open Liberty Operator to deploy the microservice to Kubernetes.
 
-First, install Custom Resource Definitions (CRDs) for the Open Liberty Operator by running the following command:
+First, install the cert-manager to your Kubernetes cluster by running the following command:
```
Comment (ramkumar-k-9286) [inline README.adoc]:
```suggestion
First, install the `cert-manager` to your Kubernetes cluster by running the following command:
```

File: README.adoc [hunk 8]
```diff
@@ -2051,118 +2063,100 @@ Tests run: 3, Failures: 0, Errors: 0, Skipped: 0
...
 [role='command']
 ```
-set OPERATOR_NAMESPACE=default
-set WATCH_NAMESPACE=\"\"
+kubectl apply --server-side -f https://raw.githubusercontent.com/OpenLiberty/open-liberty-operator/main/deploy/releases/1.2.1/kubectl/openliberty-app-crd.yaml
 ```
---
+Custom Resources extend the Kubernetes API and enhance its functionality.
+
+Set environment variables for namespaces for the Operator by running the following commands:
```
Comment (ramkumar-k-9286) [inline README.adoc]:
```suggestion
Set environment variables for namespaces for the operator by running the following commands:
```

File: README.adoc [hunk 9]
```diff
@@ -2259,14 +2253,14 @@ postgres-58bd9b55c7-6vzz8               1/1     Running   0          13s
 olo-controller-manager-6fc6b456dc-s29wl 1/1     Running   0          10m
 ----
 
-Run the following command to set up port forwarding to access the `inventory` microservice:
+Wait for a while to let the inventory service to start, then run the following command to set up port forwarding to access the `inventory` microservice:
```
Comment (ramkumar-k-9286) [inline README.adoc]:
```suggestion
Pause briefly to give the inventory service time to initialize. After it has started, use the following command to configure port forwarding to access the `inventory` microservice:
```

File: README.adoc [hunk 10]
```diff
@@ -2436,7 +2430,7 @@ endif::[]
 
 === Tearing down the environment 
 
-When you're finished trying out the microservice, press **CTRL+C** in the command line session where you ran the `kubectl port-forward` command to stop the port forwarding. You can delete all Kubernetes resources by running the `kubectl delete` commands:
+When you're finished trying out the microservice, press `CTRL+C` in the command line session where you ran the `kubectl port-forward` command to stop the port forwarding. You can delete all Kubernetes resources by running the `kubectl delete` commands:
```
Comment (ramkumar-k-9286) [inline README.adoc]: just to confirm is it `CTRL+C` or is it `CTRL+c` or `ctrl+c` ?

File: README.adoc [hunk 11]
```diff
@@ -2334,7 +2328,7 @@ include::finish/module-kubernetes/src/main/liberty/config/server.xml[]
 
 You can modify the inventory deployment to customize the service. Customizations for a service include changing the port number, changing the context root, and passing confidential information by using Secrets. 
 
-The [hotspot=contextRoot file=0]`default.context.root` variable is defined in the [hotspot file=0]`server.xml` configuration file. The context root for the inventory service can be changed by using this variable. The value for the [hotspot=contextRoot file=0]`default.context.root` variable can be defined in a ConfigMap and accessed as an environment variable.
+The [hotspot=contextRoot file=0]`context.root` variable is defined in the [hotspot file=0]`server.xml` configuration file. The context root for the inventory service can be changed by using this variable. The value for the [hotspot=contextRoot file=0]`context.root` variable can be defined in a ConfigMap and accessed as an environment variable.
```
Comment (ramkumar-k-9286) [inline README.adoc]:
```suggestion
The [hotspot=contextRoot file=0]`context.root` variable is defined in the [hotspot file=0]`server.xml` configuration file. The context root for the inventory service can be changed by using this variable. The value for the [hotspot=contextRoot file=0]`context.root` variable can be defined in a `ConfigMap` and accessed as an environment variable.
```

File: README.adoc (hunk 10 above)
Comment (ramkumar-k-9286) [inline README.adoc]:
```suggestion
When you're finished trying out the microservice, press `CTRL+C` in the command-line session where you ran the `kubectl port-forward` command to stop the port forwarding. You can delete all Kubernetes resources by running the `kubectl delete` commands:
```

File: README.adoc (hunk 3 above)
Comment (gkwan-ibm) [reply README.adoc:202]: actually is `c`. Because used `C` across all guides, we will update together if need.

File: README.adoc (hunk 10 above)
Comment (gkwan-ibm) [reply README.adoc]: actually is c. Because used C across all guides, we will update together if need.

File: README.adoc (hunk 4 above)
Comment (gkwan-ibm) [reply README.adoc]: I updated to
> ... the new `http.port` and `https.port` variables ...

File: README.adoc (hunk 6 above)
Comment (gkwan-ibm) [reply README.adoc]: I updated "{kube}" to "Kubernetes".

File: README.adoc (hunk 8 above)
Comment (gkwan-ibm) [reply README.adoc]: I updated to "the Open Liberty Operator"

**Review by ramkumar-k-9286 — APPROVED** [review]

**Pull request merged**